File: notification-schema-java/src/main/java/uk/gov/defra/tracesx/notificationschema/representation/Address.java
package uk.gov.defra.tracesx.notificationschema.representation;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Builder
@Getter
@Setter
@JsonInclude(Include.NON_EMPTY)
@NoArgsConstructor(access = AccessLevel.PUBLIC)
@AllArgsConstructor(access = AccessLevel.PUBLIC)
public class Address {

  private String street;
  private String city;
  private String country;
  private String postalCode;
}

File: imports-frontend-entities/src/entities/address.js
const handler = require('./base/handler')

module.exports = class Address {

  constructor(obj = {}) {

    this.street = obj.street
    this.city = obj.city
    this.country = obj.country
    this.postalCode = obj.postalCode

    return Object.seal(new Proxy(this, handler))
  }
}